runhouse/resources/secrets/secret.py

```diff
@@ -47,6 +47,17 @@ def __init__(
     def values(self):
         return self._values
 
+    @property
+    def config_for_rns(self):
```

Comment:
@carolineechen do we still need this? if we are removing it from the config later on before sending the request to Den

Reply:
I added this back to better handle in-memory secrets, which aren't associated with a file or env vars. (1) for reconstructing a resource, the values will be lost and won't allow for `Secret.from_config(secret.config_for_rns)` (2) simplifies the logic for `put_resource` on a cluster, works directly rather than needing to pull secrets from vault

Reply:
ah ok makes sense!

Reply:
(2) is easily addressable, but (1) seems like a property of resource that ideally we can maintain. the resource test `test_from_config` will fail if we don't have "values" in the config